internal/provider/resource_gitlab_group_saml_link_test.go

```diff
@@ -0,0 +1,189 @@
+//go:build acceptance
+// +build acceptance
+
+package provider
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/xanzy/go-gitlab"
+)
+
+func TestAccGitlabGroupSamlLink_basic(t *testing.T) {
+	rInt := acctest.RandInt()
+	resourceName := "gitlab_group_saml_link.foo"
+
+	// PreCheck runs after Config so load test data here
+	var samlLink gitlab.SAMLGroupLink
+	testSamlLink := gitlab.SAMLGroupLink{
+		Name: "test_saml_group",
+	}
+
+	resource.ParallelTest(t, resource.TestCase{
```

Comment:
Would you mind checking out [this issue here](https://github.com/gitlabhq/terraform-provider-gitlab/issues/1008) and follow the suggestions there? 

Especially having the configs inlined, the groups created outside of the configs and relying on imports between steps to verify non-computed values.

Reply:
@timofurrer do you have an example of where this has been done? I think I managed recommendations 1 and 3 from the post, but I'm a little unclear on 2. Mostly because I've never done go, tf, or provider development before. Any guidance or examples would be appreciated.